File: algorithms/leetcode/longest_consecutive_sequence.py

def longest_consecutive_sequence(numbers):
    if len(numbers) == 0:
        return 0
    s = set()
    for num in numbers:
       s.add(num)

    max_sequence = 1 
    for num in s:
        max_copy = 1
        if num -1 in s:
            continue
        num +=1
        while True:
            if num in s:
                max_copy +=1
                num +=1
            else:
                max_sequence = max(max_copy, max_sequence)
                break

    return max_sequence 

print(longest_consecutive_sequence([0,3,7,2,5,8,4,6,0,1]))
            
            

File: algorithms/leetcode/longest_consecutive_sequence.go
package main
import "fmt"

func max(a, b int) int {
  if a > b {
    return a
  }
  return b
}

func longest_consecutive_sequence(numbers []int) int {
  if len(numbers) == 0 {
    return 0
  }
  m := make(map[int]bool, len(numbers))
  for _, number := range numbers {
    if _, ok := m[number]; ok {
      continue
    }
    m[number] = true
  }
  max_sequence := 1
  for number := range m {
    mini_max := 1 
    if _, ok := m[number-1]; ok {
      continue
    }
    for {
      number +=1
      if _, ok := m[number]; ok {
        mini_max +=1 
      } else {
        max_sequence = max(max_sequence, mini_max)
        break
      }
    }
  }
  return max_sequence 
}

func main() {
  nums := []int{100, 4, 200, 1, 3, 2} 
  fmt.Println(longest_consecutive_sequence(nums))

}
